src/SonarScanner.MSBuild.Common/AnalysisProperties/EnvScannerPropertiesProvider.cs

```diff
@@ -48,7 +46,7 @@ public static bool TryCreateProvider(ILogger logger, out IAnalysisPropertyProvid
                 provider = new EnvScannerPropertiesProvider(Environment.GetEnvironmentVariable(ENV_VAR_KEY));
                 return true;
             }
-            catch (Exception ex) when (ex is JsonException || ex is IOException)
```

Comment:
JsonException is in Newtonsoft 12 :| 

hence failures on .NET Core 2.1

🤦

Reply:
Would it make sense to leave a comment here, so it will be fixed once NET 2.1 version is removed?

Reply:
I'll do a separate PR to avoid postponing merging this one.

I think there's multiple things: 
- filtering down the exception
- general cleanup after removing .NET 2.1

Reply:
https://github.com/SonarSource/sonar-scanner-msbuild/pull/1258